### ISSUE TITLE
Cancel a PR test job in queue if more commits are made on the source branch and new PR test job is triggered.

### ISSUE THREAD

**Issue by sam1111** [janinko/ghprb#440]
Hi,

Is it possible to achieve the following:
1. I have a PR test job triggered as soon as a PR is created from a source branch (say hotfix) to target branch (say main)
2. Now lets say I make a commit to hotfix on this another PR test job is triggred and it is in the queue.
3. I realize that ohh I missed this in my previous commit so I make another commit on the hotfix branch. Now what happens is that another PR test job is triggered and placed on the queue. 

Is it possible somehow to cancel the PR test job created in step 2 automatically when step 3 is executed?

**Comment by Puneeth-n** [comment]
@sam1111 This option is already available at the job level and global config level. At the job level, it is found under `Trigger Setup`. Select `Cancel build on update`. In `configure system`, it is found under `GitHub Pull Request Builder -> Application Setup -> Cancel build on update`

**Comment by polavishnu4444** [comment]
even after setting this the cancellation is not happening.

**Comment by perguth** [comment]
For me neither.

**Comment by Puneeth-n** [comment]
@polavishnu4444 @perguth Unfortunately, I cannot help you anymore as we moved away from this plugin in favor of Jenkins pipeline.

**Comment by perguth** [comment]
Thanks for letting us know. Could you sketch out how you get those two features (build on commit / abort on update)?

**Comment by Puneeth-n** [comment]
@perguth you mean in Jenkins pipelines? I'm afraid I cannot share the code now. But we are planning to open source part of our pipeline soon.  Well, we use multi branch pipeline. so Jenkins creates a new job for each PR. There is a small groovy code which checks for all running jobs and terminates them before starting the latest build. 

I have been using this method in `production` and it seems to work. However, the recommended way in pipelines is to use `lock`s and `milestone`.  The link to [locks and milestone](https://jenkins.io/blog/2016/10/16/stage-lock-milestone/)
